lambda/split_cuboids_lambda.py

```diff
@@ -0,0 +1,129 @@
+# Lambda that fanouts Index.FanoutIdWriters based on the number of unique ids
+# found in the cuboid.  Ids divided among multiple workers because step
+# function data passed between states must be less than 32K.
+#
+# It expects to get from events dictionary
+# {
+#   'config': {'kv_config': {...}
+#              'state_config': {...},
+#              'object_store_config': {...}},
+#   'cuboid_ids_bucket': '...',
+#   'id_index_step_fcn': '...', # arn of step function
+#   'fanout_id_writers_step_fcn': '...',    # step func arn
+#   'cuboid_object_key': '...',
+#   'num_ids': int,             # Number of unique ids in cuboid.
+#   'ids_s3_key': ...,          # S3 key where ids are stored for this cuboid.
+#   'version': '...',
+#   'max_write_id_index_lambdas': int,
+#   'finished': '...', # bool
+#   'fanout_params': {
+#       "max_concurrent": int,
+#       "rampup_delay": int,
+#       "rampup_backoff": float,
+#       "status_delay": int
+#   }
+# }
+#
+# Outputs (modified or added):
+# {
+#   'finished': bool    # Whether done starting Index.FanoutIdWriter step functions.
+#   'num_ids': 0,       # Zeroed to avoid infinite fanning out.
+#   'wait_secs': int    # Tell subsequent wait state how long to wait.
+#   'sub_sfn': ...
+#   'sub_args': [...]
+#   'max_concurrent': ...,
+#   'rampup_delay': ...,
+#   'rampup_backoff': ...,
+#   'status_delay': ...
+#   'running': [...]
+#   'results': [...]
+# }
+
+from cloudwatchwrapper import get_lambda_execs_in_last_min
+from heaviside.activities import fanout_nonblocking
+from math import ceil
+from random import randint
+
+NUM_IDS_PER_WORKER = 100
+
+def handler(event, context):
+
+    num_ids = int(event['num_ids'])
+    if num_ids == 0:
+        if 'sub_args' not in event or len(event['sub_args']) == 0:
+            # Nothing left to do.
+            output = {'finished': True}
+            return output
+
+    fanout_args = get_fanout_args(event)
+
+    # Change this key?
```

Comment:
comment is confusing. Is this a todo?

Reply:
Yeah, a potential todo if I start reserving lambda executions for each lambda involved in indexing.